apps/client/src/app/core/data/localized-lazy-data.service.ts

```diff
@@ -0,0 +1,463 @@
+import { LazyDataProviderService } from './lazy-data-provider.service';
+import { lazyFilesList } from './lazy-files-list';
+import { mapIds } from './sources/map-ids';
+import { zhWorlds } from './sources/zh-worlds';
+
+@Injectable({ providedIn: 'root' })
+export class LocalizedLazyDataService {
+  private readonly dataKeys = new Set(Object.keys(lazyFilesList));
+  private readonly indentRegexp = new RegExp('<Indent/>', 'i');
+
+  constructor(private lazyDataProvider: LazyDataProviderService) {}
+
+  private getResolver(key: LazyDataKey, accessor: number | string, lang: Language): Observable<string | undefined> {
+    return this.lazyDataProvider.getLazyData(key).pipe(
+      map((data) => {
+        if (!get(data, accessor)) console.warn(`Lazy data could not be resolved for key [${key}], accessor [${accessor}]`);
+        const baseValue = get(data, `${accessor}.${lang}`, undefined);
+        if (lang === 'fr') return baseValue?.replace(this.indentRegexp, '');
+        return baseValue;
+      })
+    );
+  }
+
+  private getFallbackResolver(extKey: LazyDataKey | undefined, fallbackKey: LazyDataKey, accessor: number | string, lang: Language) {
+    const fallbackResolver = this.getResolver(fallbackKey, accessor, lang).pipe(
+      switchMap((val) => (val ? of(val) : this.getResolver(fallbackKey, accessor, 'en')))
+    );
+    return of(extKey).pipe(
+      switchMap((key) => {
+        if (!key) return fallbackResolver;
+        return this.getResolver(extKey, accessor, lang).pipe(switchMap((val) => (val ? of(val) : fallbackResolver)));
+      })
+    );
+  }
+
+  private getRow(key: LazyDataKey, accessor: number | string): I18nNameLazy {
+    const { koKey, ruKey, zhKey } = this.guessExtendedLanguageKeys(key);
+
+    return {
+      de: this.getResolver(key, accessor, 'de'),
+      en: this.getResolver(key, accessor, 'en'),
+      fr: this.getResolver(key, accessor, 'fr'),
+      ja: this.getResolver(key, accessor, 'ja'),
+      ko: this.getFallbackResolver(koKey, key, accessor, 'ko'),
+      ru: this.getFallbackResolver(ruKey, key, accessor, 'ru'),
+      zh: this.getFallbackResolver(zhKey, key, accessor, 'zh'),
+    };
+  }
+
+  private guessExtendedLanguageKeys(key: LazyDataKey): ExtendedLanguageKeys {
+    return {
+      zhKey: this.guessExtendedLanguageKey('zh', key),
+      koKey: this.guessExtendedLanguageKey('ko', key),
+      ruKey: this.guessExtendedLanguageKey('ko', key),
+    };
+  }
+
+  private guessExtendedLanguageKey(language: string, key: string): LazyDataKey | undefined {
+    const guessKey = `${language}${key.charAt(0).toUpperCase()}${key.substr(1)}`;
+    if (this.isLazyDataKey(guessKey)) {
+      return guessKey;
+    }
+
+    return undefined;
+  }
+
+  public getWorldName(world: string): I18nNameLazy {
+    const i18nName: I18nNameLazy = {
+      fr: of(world),
+      en: of(world),
+      de: of(world),
+      ja: of(world),
+      zh: of(zhWorlds[world]),
+      ko: of(undefined),
+      ru: of(undefined),
```

Comment:
Because this is about world name label, I'd rather not show undefined, but get the base value instead.

```suggestion
      ko: of(world),
      ru: of(world),
```